packages/@webex/plugin-cc/src/cc.ts

```diff
@@ -120,38 +74,57 @@ export default class ContactCenter extends WebexPlugin implements IContactCenter
     };
 
     try {
-      await this.webSocket.subscribeAndConnect({
-        datachannelUrl,
-        body: connectionConfig,
-      });
-      this.$webex.logger.info('Successfully connected and subscribed.');
+      return this.httpRequest
+        .subscribeNotifications({
+          body: connectionConfig,
+        })
+        .then(async (data: WelcomeEvent) => {
+          const agentId = data.agentId;
+          const agentConfig = new AgentConfig(agentId, this.$webex, this.httpRequest);
+          this.agentConfig = await agentConfig.getAgentProfile();
+          this.$webex.logger.log(`file: ${CC_FILE}: agent config is fetched successfully`);
+
+          return this.agentConfig;
+        })
+        .catch((error) => {
+          throw error;
+        });
     } catch (error) {
-      this.$webex.logger.info(`Error connecting and subscribing: ${error}`);
-      reject(error);
+      this.$webex.logger.error(`file: ${CC_FILE}: Error during register: ${error}`);
+
+      throw error;
     }
   }
 
-  private handleEvent(eventName: string, result: EventResult) {
-    const handler = this.eventHandlers.get(eventName);
-    if (handler) {
-      clearTimeout(handler.timeoutId);
-      handler.resolve(result);
-      this.eventHandlers.delete(eventName);
+  /**
+   * This is used for agent login.
+   * @param options
+   * @returns Promise<StationLoginSuccess>
+   * @throws Error
+   */
+  public async stationLogin(data: AgentLoginRequest): Promise<StationLoginResponse> {
+    let callingSDKRegister: Promise<void> | null = null;
+
+    if (data.loginOption === LoginOption.BROWSER) {
+      this.webRTCCalling = new WebRTCCalling(this.$webex, {}); // TODO: add callingClientConfig
```

Comment:
Why is this a TODO? How is webrtc registration working without this config being passed

Reply:
addressed